crates/fuel-core/src/service/genesis/exporter.rs

```diff
@@ -174,13 +181,20 @@ where
 
         let db = db_picker(self).clone();
         let prefix = prefix.map(|p| p.to_vec());
+        let progress_tracker =
+            self.multi_progress.table_reporter(None, T::column().name());
         self.task_manager.spawn(move |cancel| {
             tokio_rayon::spawn(move || {
                 db.entries::<T>(prefix, IterDirection::Forward)
                     .chunks(group_size)
                     .into_iter()
                     .take_while(|_| !cancel.is_cancelled())
-                    .try_for_each(|chunk| writer.write(chunk.try_collect()?))?;
+                    .enumerate()
+                    .try_for_each(|(index, chunk)| {
+                        progress_tracker.set_progress(index);
```

Comment:
But it might not advance the progress. `set_progress` sets the progress to whatever you give it, even if you give it a lower value than previously. Propagates to `set_position` on the inner bar.

Simpler than keeping a counter inside of the progress, especially since we always seem to know the progress outside -- i.e. we have to keep track of the group number due to the need to update the cursors.

I'll add some docs to make that clearer.

Reply:
It is up to you, but for me, keeping the counter inside and advancing it looks more natural. In this case, you guarantee that the progress tracker can't emit the same progress twice. Plus the type is self-contained in this case.